src/battle/area_flo/5B28D0.c

```diff
@@ -2,19 +2,128 @@
 
 #define NAMESPACE b_area_flo
 
-INCLUDE_ASM(s32, "battle/area_flo/5B28D0", func_80219420_5B28D0);
+extern char* D_80233564_5CCA14[];
+extern char* D_80233870_5CCD20[];
+extern char* D_80233950_5CCE00[];
+extern char* D_80231678_5CAB28[];
+extern char* D_80231B08_5CAFB8[];
+extern char* D_802328D8_5CBD88[];
+extern char* D_80232BE8_5CC098[];
+extern char* D_802311D8_5CA688[];
+extern char* D_80231FA8_5CB458[];
+extern char* D_80232428_5CB8D8[];
 
-INCLUDE_ASM(s32, "battle/area_flo/5B28D0", func_802194E0_5B2990);
+ApiStatus func_80219420_5B28D0(ScriptInstance* script, s32 isInitialCall) {
+    s32 var1 = 0;
+    u32 counter;
 
-INCLUDE_ASM(s32, "battle/area_flo/5B28D0", func_802195A0_5B2A50);
+    for (counter = 0; counter < 11; counter++) {
```

Comment:
This applies to all instances of this function:

* `counter` should probably just be `i`
* for 11, you can do `ARRAY_COUNT(D_802311D8_5CA688)` for example. You'll need to specify the size of each array in the extern as well.
* 0 and 1 should probably be FALSE and TRUE since it looks like tihs function is passing a bool to fx 82

Reply:
Agree with what ethan said except perhaps maybe the last bullet point; we have no idea what kinds of args these fx funcs take yet

Reply:
yeah fair enough, in the context of the effect func it might not be a bool